File: scm-java/src/main/scheme/primitives/PrimitiveSysUserName.java
package scheme.primitives;

import scheme.*;

public class PrimitiveSysUserName extends Primitive {
    @Override
    public String name() {
        return "sys-user-name";
    }

    @Override
    public String info() {
        return "Syntax: (sys-user-name)\n" +
               "Library: (scm system)\n" +
               "Description: Returns the name of the currently logged-in user as a string.\n" +
               "Example:\n" +
               "  (sys-user-name) => \"alice\"";
    }
    
    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 0, 0);
        return System.getProperty("user.name").toCharArray();
    }
}

File: scm-csharp/scheme/primitives/PrimitiveSysUserName.cs
namespace scheme;

public class PrimitiveSysUserName : Primitive
{
    public override string Name()
    {
        return "sys-user-name";
    }

    public override string Info()
    {
        return
            "Syntax: (sys-user-name)\n" +
            "Library: (scm system)\n" +
            "Description: Returns the name of the currently logged-in user as a string.\n" +
            "Example:\n" +
            "  (sys-user-name) => \"alice\"";
    }
    
    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 0, 0);
        return Environment.UserName.ToCharArray();
    }
}
